app/src/main/java/crux/bphc/cms/activities/TokenViewModel.kt

```diff
@@ -0,0 +1,223 @@
+            }
+            return
+        }
+        courseDataHandler.replaceCourses(courseList)
+
+        /* Fetch Course Content */
+        _status.postValue(PROGRESS_COURSE_CONTENT)
+        val courses = courseDataHandler.courseList
+        for (course in courses) {
+            val courseSections = courseRequestHandler.getCourseData(course) ?: continue
+            for (courseSection in courseSections) {
+                val modules: List<Module> = courseSection.modules
+                for (module in modules) {
+                    if (module.modType === Module.Type.FORUM) {
+                        val discussions = courseRequestHandler.getForumDiscussions(module.instance)
+                                ?: continue
+                        for (d in discussions) {
+                            d.forumId = module.instance
+                        }
+                        courseDataHandler.setForumDiscussions(module.instance, discussions)
+                    }
+                }
+            }
+            courseDataHandler.replaceCourseData(course.id, courseSections)
+        }
+
+        /* Fetch Site News */
+        _status.postValue(PROGRESS_SITE_NEWS)
+        val discussions = courseRequestHandler.getForumDiscussions(1) // 1 is always site news
+        if (discussions != null) {
+            for (d in discussions) {
+                d.forumId = 1
+            }
+            courseDataHandler.setForumDiscussions(1, discussions)
+        }
+
+        checkLoggedIn()
+    }
+
+    fun checkLoggedIn() {
+        if (UserAccount.isLoggedIn) {
+            _startMainActivity.postValue(SingleLiveEvent(true))
+        }
+    }
+
+    fun onLogin() {
+        /*
+            We'll just create an into a specific Moodle endpoint. The Moodle website will handle the authentication,
+            generate a token and redirect the browser to the Uri with specified schema. The browser will create an
+            intent that'll launch this activity again.
+         */
+
+        // A random number that identifies the request
+        val passport = Random().nextInt(1000).toString()
+        var builder = Urls.SSO_LOGIN_URL.buildUpon()
+        builder = builder.appendOrSetQueryParameter("passport", passport)
+        builder = builder.appendOrSetQueryParameter("urlscheme", Urls.SSO_URL_SCHEME)
+        val loginUrl = builder.toString()
+
+        // Set the launch data, we need this to verify the token obtained after SSO
+        val data = HashMap<String, String>()
+        // SITE_URL must not end with trailing /
+        data["SITE_URL"] = Urls.MOODLE_URL.toString().replace("/$".toRegex(), "")
+        data["PASSPORT"] = passport
+        MyApplication.instance.setLoginLaunchData(data)
+        val intent = Intent(Intent.ACTION_VIEW)
+        intent.data = Uri.parse(loginUrl)
+        _startIntent.postValue(SingleLiveEvent(intent))
+    }
+
+    companion object {
+        const val FETCHING_DETAILS = 0
```

Comment:
Looks like this value is no longer being used. You can get rid of it and renumber the other status

Reply:
It is being used when the user is redirected to the browser to sign in with Google though right?